batch/testResultsCollector.js

```diff
@@ -1,40 +1,52 @@
 import {XUnitParser} from './parsers/xUnitParser';
-import mongoose from 'mongoose';
-import {TestSuiteModel} from '../models/testSuiteModel';
-import {TestCaseModel} from '../models/testCaseModel';
-
+import {GithubHelper} from '../helpers/githubHelper';
+import LudwigDAO from '../database/ludwigDAO';
 
 class TestResultsCollector {
-	constructor(configuration) {
-		this.configuration = configuration;
-		mongoose.connect(configuration.mongo.uri, configuration.mongo.options);
+	constructor(ludwigConfiguration) {
+		this.configuration = ludwigConfiguration;
+		this._xUnitParser = new XUnitParser(this.configuration);
+		this._githubHelper = new GithubHelper(this.configuration);
+		this._dao = LudwigDAO;
+	}
+
+	get parser() {
+		return this._xUnitParser;
+	}
+
+	get githubHelper() {
+		return this._githubHelper;
+	}
+
+	get dao() {
+		return this._dao;
```

Comment:
Actually, I'm doing this to enable me to stub the parser when testing ... and I haven't found a way to do it differently atm. But I'm all ears if there's a better way than doing this as I'd rather not have to write this for my tests.